### PR TITLE
bugfix/users-http

### PR DESCRIPTION
**Pull request by alesiahlushakova**

### CODE REVIEW

File: src/main/java/by/itech/lab/supplier/dto/UserDto.java [hunk 1]
```diff
@@ -39,7 +39,6 @@
     @Size(min = 5, max = 254)
     private String email;
     private Role role;
-    private Address address;
+    private AddressDto addressDto;
     private CustomerDto customerDto;
-    private Warehouse warehouse;
 }
```
Comment (FortuneAwaker) [inline src/main/java/by/itech/lab/supplier/dto/UserDto.java]: empty line

File: src/main/java/by/itech/lab/supplier/service/impl/UserServiceImpl.java [hunk 2]
```diff
@@ -32,7 +34,7 @@
 
     @Override
     public Optional<UserDto> findById(Long id) {
-        return userRepository.findOneWithRolesById(id).map(userMapper::map);
+        return Optional.of(userRepository.findOneWithRolesById(id).map(userMapper::map).orElseThrow(() -> new ResourceNotFoundException("User with id=" + id + " doesn't exist")));
```
Comment (FortuneAwaker) [inline src/main/java/by/itech/lab/supplier/service/impl/UserServiceImpl.java]: too long line(required less than 120 characters)

File: src/main/java/by/itech/lab/supplier/service/impl/UserServiceImpl.java [hunk 3]
```diff
@@ -45,27 +47,27 @@
...
-                            .orElseThrow();
+                            .orElseThrow(() -> new ResourceNotFoundException("User with id=" + userDTO.getId() + " doesn't exist"));
                     userMapper.update(userDTO, existing);
                     return existing;
                 })
                 .orElseGet(() -> userMapper.map(userDTO));
-
+//        if (Objects.isNull(user.getId())) {
+//            mailService.sendMail(userDTO);
+//        }
```
Comment (FortuneAwaker) [inline src/main/java/by/itech/lab/supplier/service/impl/UserServiceImpl.java]: Do we need comment here?

File: src/main/resources/db/changelog/1.0/2020-11-10-change-user-constraint.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+      author: alesia_hlushakova
+      changes:
+        - dropNotNullConstraint:
+            columnDataType: bigint
+            columnName:  customer_id
+            tableName:  user
+        - dropNotNullConstraint:
+            columnDataType: bigint
+            columnName: warehouse_id
+            tableName: user
```
Comment (FortuneAwaker) [inline src/main/resources/db/changelog/1.0/2020-11-10-change-user-constraint.yaml]: empty line

File: src/test/java/by/itech/lab/supplier/UserServiceTest.java [hunk 5]
```diff
@@ -66,7 +66,7 @@ public void init() {
         userDto.setName("john");
         userDto.setRole(Role.ROLE_ADMIN);
         userDto.setSurname("doe");
-        userDto.setAddress(address);
+       // userDto.setAddressDto(address);
         userDto.setBirthday(LocalDate.of(1999, 11, 15));
      //   userDto.setCustomer(customerMapper.map(customer));
```
Comment (FortuneAwaker) [inline src/test/java/by/itech/lab/supplier/UserServiceTest.java]: Do we need comment here?

**Review by FortuneAwaker — CHANGES_REQUESTED** [review]

**Review by FortuneAwaker — APPROVED** [review]

**Review by IrynaZazyba — APPROVED** [review]

**Pull request merged**